src/charm.py

```diff
@@ -1,104 +1,334 @@
+        self.framework.observe(
+            self.on.opensearch_relation_changed, self._opensearch_relation_changed
+        )
+        self.framework.observe(self.on.client_relation_joined, self._client_relation_handler)
+        self.framework.observe(self.on.client_relation_changed, self._client_relation_handler)
 
-    def _on_httpbin_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API.
+    @property
+    def _peers(self) -> Optional[Relation]:
+        """Fetch the peer relation.
 
-        TEMPLATE-TODO: change this example to suit your needs.
-        You'll need to specify the right entrypoint and environment
-        configuration for your specific workload. Tip: you can see the
-        standard entrypoint of an existing container using docker inspect
+        Returns:
+            Optional[Relation]: An `ops.model.Relation` object representing the peer
+                relation.
+        """
+        return self.model.get_relation(PEER)
+
+    @property
+    def _unit_ips(self) -> List[str]:
+        """Retrieve IP addresses associated with OpenSearch peers units.
 
-        Learn more about Pebble layers at https://github.com/canonical/pebble
+        Returns:
+            List[str]: a list of private-address from OpenSearch node units.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": self.model.config["thing"]},
-                }
-            },
-        }
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", pebble_layer, combine=True)
-        # Autostart any services that were defined with startup: enabled
-        container.autostart()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
-        self.unit.status = ActiveStatus()
+        addresses = [
+            str(self._peers.data[unit].get("private-address")) for unit in self._peers.units
+        ]
+        self_address = self._unit_ip
+        addresses.append(self_address)
+        return addresses
+
+    @property
+    def _unit_ip(self) -> str:
+        """Retrieve IP addresses associated with OpenSearch unit.
+
+        Returns:
+            str: IP address of the unit
+        """
+        return str(self.model.get_binding(PEER).network.bind_address)
+
+    @property
+    def _unit_name(self) -> str:
+        """Get the unit name changing "/" to "-".
```

Comment:
Why is the name change required? It may be good to clarify it in the docstring.

Reply:
It's required because I use the unit name on the node certificate and private key.
Ex: Unit `openserarch/0` becomes `opensearch-0` and end up generating `opensearch-0.pem` and `opensearch-0-key.pem`.

Using the char `/` confuses the [pathlib](https://docs.python.org/3/library/pathlib.html) to get the `path` to the file.
I will update the docstring to make it clear.